crates/topos-tce-storage/src/fullnode/mod.rs

```diff
@@ -318,23 +312,32 @@ impl ReadStore for FullNodeStore {
 
     fn get_target_stream_certificates_from_position(
         &self,
-        position: TargetStreamPositionKey,
+        position: CertificateTargetStreamPosition,
         limit: usize,
     ) -> Result<Vec<(CertificateDelivered, CertificateTargetStreamPosition)>, StorageError> {
-        let starting_position = position.2;
+        let starting_position = position.position;
+        warn!("Starting position {:?}", starting_position);
+        let prefix = TargetSourceListKey(position.target_subnet_id, position.source_subnet_id);
+        warn!("Prefix is: {:?}", prefix);
+        warn!(
+            "target_stream: {:?}",
+            self.index_tables.target_streams.iter()?.collect::<Vec<_>>()
+        );
         let x: Vec<(CertificateId, CertificateTargetStreamPosition)> = self
             .index_tables
             .target_streams
-            .prefix_iter(&(position.0, position.1))?
-            .skip((starting_position.0).try_into().map_err(|_| {
+            .prefix_iter(&prefix)?
+            .skip(starting_position.try_into().map_err(|_| {
                 StorageError::InternalStorage(InternalStorageError::InvalidQueryArgument(
                     "Unable to parse Position",
                 ))
             })?)
             .take(limit)
-            .map(|(k, v)| (v, k.into()))
+            .map(|(k, v)| (v, k))
```

Comment:
Couldn't this be a potentially very large collection of items for a subnet with a long history?